pkg/vdri/trustbloc/selection/staticselection/service.go

```diff
@@ -18,9 +18,21 @@ func NewService() *SelectionService {
 	return &SelectionService{}
 }
 
-// SelectEndpoints select endpoints
+// SelectEndpoints select the first endpoint listed for each stakeholder
 func (ds *SelectionService) SelectEndpoints(endpoints []*endpoint.Endpoint) ([]*endpoint.Endpoint, error) {
-	// TODO add logic to select endpoints
-	// For now we just return all endpoints !!! need to be removed after adding selection logic
-	return endpoints, nil
+	var out []*endpoint.Endpoint
+
+	domains := map[string]struct{}{}
+
+	for _, ep := range endpoints {
+		// skip endpoints whose domains are already listed
+		if _, present := domains[ep.Domain]; present {
```

Comment:
you need Round Robin

Reply:
Does random selection now - round robin would require state which remembers the list of endpoints belonging to each stakeholder, not sure how/whether to do that in the selection service.

Reply:
@bstasyszyn ^^^

Reply:
@Moopli https://github.com/hyperledger/fabric-sdk-go/blob/cb4ad138c14fd98039c56ab6301e665254a930bc/pkg/client/common/selection/dynamicselection/pgresolver/lbpolicyimpl.go

Reply:
I think random is good enough for static selection. We'll need to have a dynamic implementation of selection which can have all the bells and whistles, including pluggable load balancers/priority selectors.